pkg/cosign/verify.go

```diff
@@ -928,6 +947,58 @@ func VerifyBundle(ctx context.Context, sig oci.Signature, rekorClient *client.Re
 	return true, nil
 }
 
+func VerifyTSABundle(ctx context.Context, sig oci.Signature, tsaClient *tsaclient.TimestampAuthority, tsaCertChainPath string) (bool, error) {
+	bundle, err := sig.TSABundle()
+	if err != nil {
+		return false, err
+	} else if bundle == nil {
+		return false, nil
+	}
+
+	b64Sig, err := sig.Base64Signature()
+	if err != nil {
+		return false, fmt.Errorf("reading base64signature: %w", err)
+	}
+
+	cert, err := sig.Cert()
+	if err != nil {
+		return false, err
+	}
+
+	sigBytes, err := base64.StdEncoding.DecodeString(b64Sig)
+	if err != nil {
+		return false, fmt.Errorf("reading DecodeString: %w", err)
+	}
+	pemBytes, err := os.ReadFile(filepath.Clean(tsaCertChainPath))
+	if err != nil {
+		return false, fmt.Errorf("error reading certification chain path file: %w", err)
+	}
+	certPool := x509.NewCertPool()
```

Comment:
Can you add a TODO to make sure we update this logic once we finish https://github.com/sigstore/timestamp-authority/issues/121, because this isn't a sufficient check (since this sets the leaf as a the "trusted" cert)

Reply:
Done